server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1846,4 +1846,36 @@ public void test_generated_sub_column_provided_null_is_validated() {
         execute("insert into t (id, obj) (select 1, {\"a\" = {\"b\" = null}} from sys.cluster)");
         assertThat(response.rowCount()).isEqualTo(0L);
     }
+
+    @Test
+    public void test_generated_column_can_refer_to_a_non_string_partitioned_by_column() {
+        execute("""
+            create table t(a int, parted int, gen_from_parted int as parted +1) partitioned by (parted);
+            """
+        );
+
+    }
+
+    @Test
+    public void test_non_string_partitioned_by_column_can_be_used_in_expressions() {
+        execute("""
+             CREATE TABLE t (
+                 a INT,
+                 parted INT CHECK (parted > 1),
+                 gen_from_parted INT as parted + 1
+             ) PARTITIONED BY (parted)
+             """
+        );
+
+        // Partitioned column value used to be provided as "2" internally.
+        // It caused casting error and thus:
+        //  - record with passing CHECK was rejected.
+        //  - generated expression couldn't access partitioned column value.
+        execute("insert into t (a, parted) select 1, 2");
+        assertThat(response.rowCount()).isEqualTo(1L);
+        refresh();
+
+        execute("select a, parted, gen_from_parted from t");
+        assertThat(response).hasRows("1| 2| 3");
+    }
```

Comment:
It's not a regression on 5.3.0. 
Checked both scenarios on 5.2 branch.

gen expression: record rejected with `ClassCastException` in `InsertSourceFromCells`, somewhere in `IntegerType.sanitizeValue` 

 CHECK  record rejected with  `"left and right must have the same type for comparison"`.
 
I think I can drop this test?

Reply:
Sorry, didn't understand the question. I copied test into 5.2 branch and slightly modified to run each case separately:

CHECK:
```
CREATE TABLE t (
      a INT,
      parted INT CHECK (parted > 1)
) PARTITIONED BY (parted)
```
`insert into t (a, parted) select 1, 2`

Fails with `AssertionError: left and right must have the same type for comparison`. Not rejected but fails, sorry was not clear in the first message.

GEN:
```
CREATE TABLE t (
     a INT,
     parted INT, ==> no check this time
     gen_from_parted INT as parted + 1
 ) PARTITIONED BY (parted)
```

`insert into t (a, parted) select 1, 2` - insert 0 rows. In debugger error for item is ClassCastException.

What I'm saying is that we used different components for ingestion in 5.2 but appretnyl they had same casting issues for partitioned columns.

Reply:
Thanks for the clarification. So I think the indexer test looks enough to me